src/openrct2-ui/windows/NewRide.cpp

```diff
@@ -698,6 +741,55 @@ class NewRideWindow final : public Window
         return nextListItem;
     }
 
+    bool IsFiltered(rct_ride_entry* rideEntry)
+    {
+        if (Filter.empty())
+            return true;
+
+        return IsFilterInRideType(rideEntry) || IsFilterInRideName(rideEntry) || IsFilterInIdentifier(rideEntry)
+            || IsFilterInAuthors(rideEntry) || IsFilterInFilename(rideEntry);
+    }
+
+    bool IsFilterInRideType(rct_ride_entry* rideEntry)
```

Comment:
Please use `const rct_ride_entry&` here and in the other functions you add. We’re trying to move away from pointers in function signatures.

Reply:
Using `const rct_ride_entry&` is a little troublesome because the call to `get_ride_naming` in `ISFilterInRideType` takes a `rct_ride_entry*`. I could make a copy and pass that instead or I could just remove the `const` and use `rct_ride_entry&` instead. I'm not sure what your preference is though.

Reply:
It’s fine, it’s not that intrusive.